chainermn/dataset.py

```diff
@@ -15,6 +16,8 @@ def scatter_dataset(dataset, comm):
         dataset: A dataset (e.g., ``list``, ``numpy.ndarray``,
             ``chainer.datasets.TupleDataset``, ...).
         comm: ChainerMN communicator or MPI4py communicator.
+        shuffle: Shuffle the dataset before being scattered.
+        root: The root process of the scatter operation.
```

Comment:
Add explanation.
 * I checked Chainer's dataset.py and the sentence of ``shuffle`` is improved.
 * Added type specifications to ``shuffle`` and ``root``.
 * Added exaplanation of ``seed`` 
Thanks!